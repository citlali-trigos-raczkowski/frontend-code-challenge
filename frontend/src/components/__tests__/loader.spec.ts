import { shallowMount, createLocalVue } from "@vue/test-utils";
import Loader from "../loader.vue";
import VueRouter from "vue-router";
import { ExpectedLoaderData } from "../../../tests/mocks/expected-data";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("component/loader.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(Loader, {
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("includes expected values", () => {
    const wrapper = shallowMount(Loader, {
      localVue,
    });
    const aWrapper = wrapper.find("v-progress-circular-stub");

    expect(aWrapper.attributes().size).toBe(ExpectedLoaderData.size);
    expect(aWrapper.attributes().width).toBe(ExpectedLoaderData.width);
    expect(aWrapper.attributes().indeterminate).toBe(
      ExpectedLoaderData.indeterminate
    );
    expect(aWrapper.attributes().color).toBe(ExpectedLoaderData.color);
  });
});
