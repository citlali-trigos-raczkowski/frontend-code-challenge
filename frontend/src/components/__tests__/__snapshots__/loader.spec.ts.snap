// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/loader.vue renders correctly 1`] = `
<div
  id="loader"
>
  <v-progress-circular-stub
    color="amber"
    indeterminate="true"
    rotate="0"
    size="70"
    value="0"
    width="7"
  />
</div>
`;
